framework/src/main/java/org/checkerframework/framework/util/typeinference8/constraint/ConstraintSet.java

```diff
@@ -162,26 +184,38 @@ public void remove(ConstraintSet subset) {
    */
   public ConstraintSet getClosedSubset(Dependencies dependencies) {
     ConstraintSet subset = new ConstraintSet();
-    Set<Variable> inputDependencies = new LinkedHashSet<>();
-    Set<Variable> outDependencies = new LinkedHashSet<>();
+    // Collection of all outputs of the constraints in this set.
+    Set<Variable> allOutputs = new LinkedHashSet<>();
+    for (Constraint constraint : list) {
+      if (constraint instanceof TypeConstraint) {
+        allOutputs.addAll(((TypeConstraint) constraint).getOutputVariables());
+      }
+      // No other constraints have output variables
+    }
+
+    // Find a subset of this set where the following is true for all the constraints in the subset:
+    // no input variable of a constraint can influence an output variable of another constraint in
+    // the subset.
+    // (Influence means that neither variable can depend on the other.)
     for (Constraint constraint : list) {
       if (constraint.getKind() == Kind.EXPRESSION
           || constraint.getKind() == Kind.LAMBDA_EXCEPTION
           || constraint.getKind() == Kind.METHOD_REF_EXCEPTION) {
         TypeConstraint c = (TypeConstraint) constraint;
-        Set<Variable> newInputs = dependencies.get(c.getInputVariables());
-        Set<Variable> newOutputs = dependencies.get(c.getOutputVariables());
-        if (Collections.disjoint(newInputs, outDependencies)
-            && Collections.disjoint(newOutputs, inputDependencies)) {
-          inputDependencies.addAll(newInputs);
-          outDependencies.addAll(newOutputs);
-          subset.add(c);
-        } else {
-          // A cycle (or cycles) in the graph of dependencies between constraints exists.
-          subset = new ConstraintSet();
-          break;
+        List<Variable> inputs = c.getInputVariables();
+        boolean found = false;
+        for (Variable in : inputs) {
+          for (Variable out : allOutputs) {
```

Comment:
The use of `allOutputs` here enforces a stricter condition than that documented above on lines 196-199.  It enforces that no input of a constraint in the subset can influence any output variable whatever, even of a variable that is not in the subset.  Is that intentional?

Reply:
The documentation was wrong.